cmd/kubeadm/app/apis/bootstraptoken/v1/utils.go

```diff
@@ -96,11 +96,12 @@ func NewBootstrapTokenStringFromIDAndSecret(id, secret string) (*BootstrapTokenS
 
 // BootstrapTokenToSecret converts the given BootstrapToken object to its Secret representation that
 // may be submitted to the API Server in order to be stored.
-func BootstrapTokenToSecret(bt *BootstrapToken) *v1.Secret {
+func BootstrapTokenToSecret(bt *BootstrapToken, labels map[string]string) *v1.Secret {
```

Comment:
hm, i know this function is just for kubeadm use mainly but with this change we are modifying the signature. its a breaking change in a v1 package.

Reply:
we could introduce a new function called BootstrapTokenToSecretWithLabels

Reply:
Based on our previous discussion, maybe we shouldn't support separate calls `kubeadm init phase show-join-command`.  Because it has a strong dependence on `BootstrapToken` and `UploadCerts` phases. 
Our current workflow framework does not support dependency checking, and perhaps we need to re-evaluate whether this is necessary... @neolit123

Reply:
if we are adding the phase it makes sense to support calling it as a CLI, but i think we should just error out because the dependencies are not there. i guess the main purpose would be for users to be able to skip it

Reply:
In order to implement dependency verification,  I introduce a new field `Dependencies` for `Phase`.
I think this is fine, because it is not a break change, the original behavior will not be affected. 
What do you think? @neolit123

Reply:
If you run `kubeadm init phase show-join-command`, the error output will be something like this:
```sh
unresolved dependencies: missing "bootstrap-token" phase which is needed by "show-join-command" phase
To see the stack trace of this error execute with --v=5 or higher
```

Reply:
i think one problem with the dependencies concept is how do users execute dependencies? it is technically not possible to run two phases one after another to solve the dependencies problem. they can only skip all phases except the target phase and it's dependencies.